### PR TITLE
Create backend service with php

### LINKED ISSUE #7: Create backend service with PHP
### Clear and concise description

As a developer I want to use an API service for querying from database. To do this, you need to add easy and reusable database adapter.

Requirements

- [x] APP-2 is done
- [x] the database adapter could not be part of the application, it must be imported from separate project
- [x] the adapter must use PDO
- [x] use at least one (repository) pattern
- [x] let's deal with the most common queries (in progress, count, ...)

### Additional context

APP-9 is depending on this.

### PR DESCRIPTION
**Pull request by csaba-nagy**
### Description

In this pull request, the model-repository-controller pattern has been set, and the methods for the CRUD operations have been created.

### Linked Issues

It closes #7

### CODE REVIEW

File: packages/api-php/src/Controllers/BaseController.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  public function __construct(protected BaseRepository $repository)
+  {
+  }
+
+  public function index(): ?array
+  {
+    return $this->repository->findAll();
+  }
+
+  public function show(int $id)
```
Comment (nandordudas) [inline packages/api-php/src/Controllers/BaseController.php:20]: Missing return type

File: packages/api-php/src/Database/Database.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+declare(strict_types=1);
+
+namespace TodoApp\ApiPhp\Database;
+
+use DatabaseAdapterPhp\Abstractions\AbstractDatabaseConnector;
+use DatabaseAdapterPhp\Factories\PostgreSQLConnectionFactory;
+
+class Database
+{
+  private static ?AbstractDatabaseConnector $connection;
```
Comment (nandordudas) [inline packages/api-php/src/Database/Database.php:12]: Interface should be better than abstract class

File: packages/api-php/src/Database/Database.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  // TODO: use environment variables to set the connection data
+  private const CONNECTION_DATA = null;
+  private const PDO_OPTIONS = null;
+
+  public static function getConnection()
+  {
+    return self::$connection
```
Comment (nandordudas) [inline packages/api-php/src/Database/Database.php:20]: 🤘

File: packages/api-php/src/Contracts/ModelInterface.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace TodoApp\ApiPhp\Contracts;
+
+interface ModelInterface
+{
+  public function findAll(): ?array;
+  public function create(array $payload): array;
+  public function getById(int $id): array;
+  public function update(array $payload): array;
+  public function delete(int $id): array;
```
Comment (nandordudas) [inline packages/api-php/src/Contracts/ModelInterface.php:13]: What's the reason of returning with array?

**Review by nandordudas — APPROVED** [review]
Good idea and solution 💪

File: packages/api-php/src/Controllers/BaseController.php (hunk 1 above)
Comment (csaba-nagy) [reply packages/api-php/src/Controllers/BaseController.php:20]: Indeed, thank you!

File: packages/api-php/src/Database/Database.php (hunk 2 above)
Comment (csaba-nagy) [reply packages/api-php/src/Database/Database.php:12]: It has been modified, thank you!

File: packages/api-php/src/Contracts/ModelInterface.php (hunk 4 above)
Comment (csaba-nagy) [reply packages/api-php/src/Contracts/ModelInterface.php]: The create, getById and update methods return the data as an associative array. The delete method return type is wrong, it changed to void.

Comment (nandordudas) [reply packages/api-php/src/Contracts/ModelInterface.php]: Nice fix.

File: packages/api-php/src/Models/BaseModel.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+abstract class BaseModel implements ModelInterface
+{
+  abstract public function findAll(): ?array;
+  abstract public function create(array $payload): array;
+  abstract public function getById(int $id): array;
+  abstract public function update(array $payload): array;
+  abstract public function delete(int $id): void;
+
+  public function __construct(protected AbstractDatabaseConnector $database)
```
Comment (nandordudas) [inline packages/api-php/src/Models/BaseModel.php:18]: Please use interface instead - `Connectable` - of using `AbstractDatabaseConnector`.

It is more convenient, meaningful and wide specific.

**Review by nandordudas — APPROVED** [review]

File: packages/api-php/src/Models/BaseModel.php (hunk 5 above)
Comment (csaba-nagy) [reply packages/api-php/src/Models/BaseModel.php:18]: Okay, I modified as you requested.

**Pull request merged**